### PR TITLE
Photo grid and modal

### PR DESCRIPTION
**Pull request by christiewarren**

### CODE REVIEW

File: fam-archive-app/src/components/PhotoGrid.js [hunk 1]
```diff
@@ -67,23 +67,14 @@ export default function PhotoGrid(){
     }
 
     function checkForFirstLastImage(selectedIndex){
```
Comment (mrmancini) [inline fam-archive-app/src/components/PhotoGrid.js]: I would name this setArrowButtonsDisabled instead. checkForFirstLastImage sounds like it returns a boolean and doesn't have side effects

File: fam-archive-app/src/App.css [hunk 2]
```diff
@@ -213,16 +213,25 @@ a{
...
-  font-size: 16px;
-  padding: 10px;
+  font-size: 20px;
+  padding: 17px 18px 14px;
   margin: 15px;
   cursor: pointer;
+  box-shadow: 0px 1px 14px 0px rgba(0, 0, 0, 0.35);
+}
+
+#prev-photo-button > img, #next-photo-button > img{
```
Comment (mrmancini) [inline fam-archive-app/src/App.css]: You could use :is here
```css
:is(#prev-photo-button, #next-photo-button) > img
```

File: fam-archive-app/src/App.css [hunk 3]
```diff
@@ -137,41 +137,44 @@ a{
...
+  max-height: 95vh;
+  max-width: 85%;
+  border-radius: 0 4px 4px 0;
 }
 
 .photo-modal-is-shown{
   visibility: visible;
 }
 
+/* flex items have initial setting of min-width: auto, so they don't shrink past the size of their content. Adding min-width: 0 allows item to shrink. Don't fully understand how this works. https://stackoverflow.com/questions/41774646/image-is-not-resizing-in-a-flexbox-layout */
```
Comment (mrmancini) [inline fam-archive-app/src/App.css:167]: Good commment

File: fam-archive-app/src/components/PhotoGrid.js [hunk 4]
```diff
@@ -11,78 +11,64 @@ export default function PhotoGrid(){
...
-        //problem: console always logs clicked when overlay is clicked, but modal doesnt disappear. Class highlights in dev mode, but doesn't turn off 
-        // console.log('clicked')
-
-        //when overlay or x is clicked: toggle modal-is-shown-class (to hide modal)
+        addOverlayListener()
     }
-    
 
     function togglePhotoModal(){
+        //toggle photo-modal-is-shown class, which changes visibility to visible (may find a better way to do this)
```
Comment (mrmancini) [inline fam-archive-app/src/components/PhotoGrid.js]: You can do
```jsx
className = {this.state.showPhotoModal ? 'photo-modal-is-shown' : ''}
```

File: fam-archive-app/src/components/PhotoGrid.js [hunk 5]
```diff
@@ -35,8 +37,19 @@ export default function PhotoGrid(){
...
-        let photoModal = document.getElementById('photo-modal')
-        photoModal.classList.toggle('photo-modal-is-shown')
+        const photoModal = document.getElementById('photo-modal')
+        setModalVisibility((prevIsModalShown) => {
+            console.log(prevIsModalShown)
+            if(prevIsModalShown){
+                photoModal.style.visibility = 'hidden'
+            }else{
+                photoModal.style.visibility = 'visible'
+            }
```
Comment (mrmancini) [inline fam-archive-app/src/components/PhotoGrid.js]: You shouldn't be modifying elements directly in React generally. Something like this
```jsx
className = {modalVisibility ? 'visible' : 'invisble'}
```
or binding to the style attribute instead.

Your code could have been shorter also
```js
photoModal.style.visibility = prevIsModalShown ? 'hidden' : 'visible'
```

**Review by mrmancini — APPROVED** [review]

File: fam-archive-app/src/components/PhotoGrid.js (hunk 5 above)
Comment (christiewarren) [reply fam-archive-app/src/components/PhotoGrid.js]: big lesson: don't use getElementById to adjust html elements. instead use state & props to directly update the JSX. the former basically defeats the point of React, and the latter ensures that the update made won't be lost upon rerender

**Pull request merged**